docs/HowToFAQ/Errors-Compiler.md

```diff
@@ -1,104 +1,529 @@
+
+To enforce a deterministic order to the evaluation, use a chain of if-then-else statements.
+
+
+## **Error: nondeterministic loop forbidden by the --enforce-determinism option**
+
+<!-- %check-run %options --enforce-determinism -->
+```dafny
+method m(b: bool) decreases * {
+  while * 
+    decreases *
+  { }
+}
+```
+
+The `--enforce-determinism` option requires a target program to be deterministic and predictable:
+there may be no program statements that have an arbitrary, even if deterministic, result.
+Hence this 'non-deterministic while' (`while *`) statement is not permitted with `--enforce-determinism`.
+
+## **Error: case-based loop forbidden by the --enforce-determinism option**
+
+<!-- %check-run %options --enforce-determinism -->
+```dafny
+method m(k: int) {
+  var i := k;
+  while 
+    decreases if i < 0 then -i else i
+  {
+    case i > 0 => i := i - 1;
+    case i < 0 => i := i + 1;
+  }
+}
+```
+
+The `--enforce-determinism` option requires a target program to be deterministic and predictable:
+there may be no program statements that have an arbitrary, even if deterministic, result.
+The case-based while statement allows the case guards to be evaluated in an arbitrary order and
+an arbitrary one of those found to be true is chosen to be executed.
+Hence the case-based loop is not permitted with `--enforce-determinism`.
+
+To enforce a deterministic order to the evaluation, use a a chain of if-then-else statements
+or series of `if` statements in which the then-branch ends in a continue statement.
+
+## **Error: compiler currently does not support assignments to more-than-6-dimensional arrays in forall statements**
+
+TODO
+
+## **Error: modify statement without a body forbidden by the --enforce-determinism option**
+
+<!-- %check-run %options --enforce-determinism -->
+```dafny
+class A { var j: int }
+
+method m(k: int, o: A) 
+  modifies o
+{
+  var i := k;
+  modify o;
+}
+```
+
+The `--enforce-determinism` option requires a target program to be deterministic and predictable:
+there may be no program statements that have an arbitrary, even if deterministic, result.
+
+The [`modify` statement](../DafnyRef/DafnyRef#sec-modify-statement) allows the program to modify 
+the state of the given objects in an arbitrary way
+(though each field still has a legal value of its declared type).
+Hence such a statement is not permitted with `--enforce-determinism`.
+
+Note that a `modify` statement with a body is deprecated.
+
+
+## **Error: this let-such-that expression is too advanced for the current compiler; Dafny's heuristics cannot find any bound for variable '_name_'**
+
+TODO
+
+## **Error: Comparison of a handle can only be with another handle**
+
+<!-- TODO -->
+_The documentation of the {:handle} attribute is in progress._
+
+<!-- DafnyCore/Compilers/Synthesizer-Csharp.cs -->
+
+## **Error: '_feature_' is not an element of the _target_ compiler's UnsupportedFeatures set**
```

Comment:
This looks like a duplicate.

Reply:
Yes. Removed. (Though in many cases of replicates the same error message is produced I more than one place and I made sure each one was reachable, even if there is just one example.)